### PR TITLE
Added Close() to DbDataReader contract. Also changed DbDataReader.Dispose() to call Close(), so as to match the Framework implementation.

### PR DESCRIPTION
**Pull request by corivera**
Ports DbDataReader.Close() changes from https://github.com/dotnet/corefx/commit/7888677b3c53596085c2fa53860d9a0519eeecfd to 1.1.0.

Fixes issue: https://github.com/dotnet/corefx/issues/2249

Shiproom approval pending.

### CODE REVIEW

**Comment by danmosemsft** [comment]
@corivera you will need to update the package version (this branch is back in the days of granular packages)

Sometihng like https://github.com/dotnet/corefx/pull/23271 is doing. @weshaggard or @Priya91 can indicate exactly where.

**Comment by danmosemsft** [comment]
Also is there a test to port also? That would sanity check the product actually picks it up (ie the versioning is done right)

**Comment by danmosemsft** [comment]
@weshaggard or @ericstj can you please review the packaging changes?

**Comment by corivera** [comment]
Looking into build failures. Probably caused by SqlClient not picking up the right System.Data.Common version

**Comment by weshaggard** [comment]
You need to add your package to the build list at https://github.com/dotnet/corefx/blob/release/1.1.0/src/packages.builds#L47.

File: src/packages.builds [hunk 1]
```diff
@@ -44,6 +44,9 @@
     <Project Include="System.IO.FileSystem.DriveInfo\pkg\System.IO.FileSystem.DriveInfo.builds">
       <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
     </Project>
+    <Project Include="System.Data.Common\pkg\System.Data.Common.builds">
```
Comment (corivera) [inline src/packages.builds]: @weshaggard Should this point to the builds file or the pkgproj?

Comment (weshaggard) [inline src/packages.builds]: The builds file.

**Comment by weshaggard** [comment]
test Innerloop Ubuntu14.04 Debug Build and Test

**Comment by corivera** [comment]
@weshaggard Still trying to verify that

**Comment by corivera** [comment]
Included System.Data.Common as a project reference in the 1.1.0 SqlClient tests, and didn't see any issues from the new virtual Close() method. I could create a custom DbDataReader class overriding Close(), but there were no errors from SqlDataReader (which implements Close() as internal).

**Comment by corivera** [comment]
Closing PR for now. Will revisit this issue in the next servicing event.

**Comment by weshaggard** [comment]
>Closing PR for now. Will revisit this issue in the next servicing event.

**Pull request closed without merging**